modules/post/windows/recon/host_info.rb

```diff
@@ -0,0 +1,149 @@
+      # elsif have_powershell?
+      #   interface = psh_exec("(Get-NetAdapter|Where-Object{$_.Name -eq (Get-NetIPAddress|Where-Object{$_.IPAddress -eq '#{session.session_host}'}).InterfaceAlias}).InterfaceDescription")
+      #   mac = psh_exec("(Get-NetAdapter|Where-Object{$_.Name -eq (Get-NetIPAddress|Where-Object{$_.IPAddress -eq '#{session.session_host}'}).InterfaceAlias}).MacAddress").gsub('-', ':')
+      else
+        # Since theres no powershell, attempt to claim it using netsh
+        interface = cmd_exec('netsh interface ip show ipaddresses').scan(/Interface \d+: ([^\r\n]+)\r\n\r\nAddr Type  DAD State   Valid Life Pref. Life Address\r\n---------  ----------- ---------- ---------- ------------------------\r\n\S+\s+\S+\s+\S+\s+\S+\s+(\d{,3}\.\d{,3}\.\d{,3}\.\d{,3})\r\n/)
+
+        # if the interface query gave anything, then check which one pertains to our IP address
+        interface = interface.find { |x| x[1] == session.session_host }[0] if interface.any?
+
+        if !interface.empty?
+          # Now that we have our interface time to get our mac address
+          mac_list = CSV.parse(cmd_exec('getmac /v /fo csv'))
+          mac = mac_list.find { |row| row[0] == interface }[2].gsub('-', ':')
+        end
+      end
+    end
+    print_good("The session is running on address #{session.session_host} (#{mac}) on interface #{interface}")
+    report_host(host: session.session_host, mac: mac) if active_db? && !mac.blank?
+  end
+
+  # Obtains information about the system architecture
+  def recon_architecture
+    if session.type == 'meterpreter'
+      host_arch = sysinfo['Architecture']
+    elsif session.type == 'powershell'
+      host_arch = case cmd_exec('(Get-ComputerInfo).OSArchitecture').strip
+                  when '64-bit'
+                    'x64'
+                  when '32-bit'
+                    'x86'
+                  end
+    else
+      host_arch = case cmd_exec('echo %PROCESSOR_ARCHITECTURE%').strip
+                  when /amd64/i || /x64/i
+                    'x64'
+                  else
+                    'x86'
+                  end
+    end
+    print_good("The hosts architecture is #{host_arch}")
+    report_host(host: session.session_host, arch: host_arch) if active_db?
+  end
+
+  # Obtain information about the OS
+  def recon_os
+    case session.type
+    when 'meterpreter'
+      host_os = sysinfo['OS']
+    when 'powershell'
+      host_os = cmd_exec('"$((Get-ComputerInfo).WindowsProductName) ($((Get-ComputerInfo).OSVersion))"').strip
+    else
+      host_data = cmd_exec('systeminfo')
+      host_os_name = host_data[/^OS Name:\s+.+$/].gsub(/^OS Name:\s+/, '').strip
+      host_os_build = host_data[/^OS Version:\s+.+$/].gsub(/^OS Version:\s+/, '').strip
+      host_os = "#{host_os_name} (#{host_os_build})"
+    end
+    print_good("The host is running #{host_os}")
+    report_host(host: session.session_host, os_name: host_os) if active_db?
+  end
```

Comment:
We have the `Msf::Post::Windows::Version` mixin which will determine the OS version information. It'd be better to use that to ensure that the module reports consistent information with what other modules are using.